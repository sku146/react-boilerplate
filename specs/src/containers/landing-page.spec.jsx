import React from 'react';
import { shallow } from 'enzyme';
import LandingPage from 'containers/landing-page';
import createRouterContext from 'react-router-test-context';

describe("Landing Page", () => {
  const ShallowComponent = (props) => {
    const context = createRouterContext();
    return shallow(<LandingPage {...props}/>).shallow({ context }).shallow();
  };

  it('component should render as expected', ()=> {
    const component = ShallowComponent().shallow();
    expect(component).toMatchSnapshot();
  });
});
