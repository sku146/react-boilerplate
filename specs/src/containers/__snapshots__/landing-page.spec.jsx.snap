// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing Page component should render as expected 1`] = `
<div
  className="landing-page"
>
  <Header
    as="h1"
    data-bdd-selector="landingHeader"
  >
    React Webpack Sample Application
  </Header>
  <Segment>
    <Header
      as="h2"
      data-bdd-selector="subHeader"
    >
      Sample UI Components
    </Header>
    <Button
      as="button"
      data-bdd-selector="primaryButton"
      primary={true}
    >
      Primary Button
    </Button>
    <Button
      as="button"
      data-bdd-selector="secondaryButton"
      secondary={true}
    >
      Secondary Button
    </Button>
  </Segment>
  <Header
    as="h2"
    data-bdd-selector="routeHeader"
  >
    Read me and Features
  </Header>
</div>
`;
